src/main/java/com/epam/esm/dao/impl/SQLTagDaoImpl.java

```diff
@@ -0,0 +1,71 @@
+package com.epam.esm.dao.impl;
+
+import com.epam.esm.dao.TagDao;
+import com.epam.esm.dao.exception.ErrorCodeEnum;
+import com.epam.esm.dao.exception.PersistenceException;
+import com.epam.esm.model.Tag;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.jdbc.support.KeyHolder;
+import org.springframework.stereotype.Repository;
+
+import java.sql.PreparedStatement;
+import java.sql.Statement;
+import java.util.List;
+
+@Repository("MySQLTagDao")
```

Comment:
Why do you need "MySQLTagDao" name here?

Reply:
In part 3 of a module it is required to transition to hibernate using names will make it easy to auto wire right repository with qualifier annotation